src/full/Agda/Interaction/Library.hs

```diff
@@ -219,15 +221,25 @@ getAgdaLibFiles' path = findProjectConfig' path >>= \case
 -- | Get dependencies and include paths for given project root:
 --
 --   Look for @.agda-lib@ files according to 'findAgdaLibFiles'.
---   If none are found, use default dependencies (according to @defaults@ file)
---   and current directory (project root).
+--   If none are found or if none of them include the current working
+--   directory, use default dependencies (according to @defaults@
+--   file) and current directory (project root).
 --
 getDefaultLibraries
   :: FilePath  -- ^ Project root.
   -> Bool      -- ^ Use @defaults@ if no @.agda-lib@ file exists for this project?
   -> LibM ([LibName], [FilePath])  -- ^ The returned @LibName@s are all non-empty strings.
 getDefaultLibraries root optDefaultLibs = mkLibM [] $ do
-  libs <- getAgdaLibFiles' root
+
+  -- We are only interested in @.agda-lib@ files that actually
+  -- include the current working directory.
+  let includesCwd lib =
+        let libBase = takeDirectory $ lib ^. libFile
+            relativeRoot = fromMaybe __IMPOSSIBLE__ $ List.stripPrefix libBase root
+        in any (`List.isPrefixOf` relativeRoot) $ lib ^. libIncludes
```

Comment:
What is `root` here? It's not clear to me whether that `List.isPrefixOf` should be `==`.

Reply:
`root` is the current working directory. Let me give a imaginary example: you try to load `~/agda-stdlib/README/Design/Decidability.agda`. In this case, the `root` is `"~/agda-stdlib/README/Design"`, the `libBase` is `"~/agda-stdlib"` (since that is the directory containing the `.agda-lib` file), the `relativeRoot` is `"README/Design/Decidability.agda"`, and `lib ^. libIncludes` is `["src"]`. We chec that `"src"` is a prefix of "README/Design/Decidability.agda", which is it not, so we default back to `-i.` instead.

Reply:
`isPrefixOf` makes me uneasy. Can we use a filepath-specific function
instead of relying on the fact both are relative and/or absolute?

(more generally it would be nice to move away from `type FilePath = String`)

Reply:
Just as a comment for a future research project, it may be interesting to think of extending Agda with relative and absolute URLs and think of the local versions as in ~/.agda/stdd-lib as caches of remote versions named by URLs. 
As a guide one can think of how when writing html files locally one uses relative URLs to refer to content before publication on one's file system, and absolute URLs for remote content. One can then improve that with something akin to a browser cache for content that is published remotely but accessed locally without making a connection to the web.